subsys/bluetooth/controller/ll_sw/ull_filter.c

```diff
@@ -996,3 +1097,143 @@ static void filter_clear(struct lll_filter *filter)
 	filter->enable_bitmask = 0;
 	filter->addr_type_bitmask = 0;
 }
+
+#if defined(CONFIG_BT_CTLR_SW_DEFERRED_PRIVACY)
+static void target_resolve(struct k_work *work)
+{
+	u8_t j, idx;
+	bt_addr_t *search_rpa;
+	struct target_resolve_work *twork;
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, 0, NULL};
+
+	twork = CONTAINER_OF(work, struct target_resolve_work, target_work);
+	idx = twork->idx;
+	search_rpa = &(twork->rpa);
+
+	if (rl[idx].taken && !bt_addr_cmp(&(rl[idx].target_rpa), search_rpa)) {
+		j = idx;
+	} else {
+		/* No match - so not in list Need to see if we can resolve */
+		if (bt_rpa_irk_matches(rl[idx].local_irk, search_rpa)) {
+			/* Could resolve, store RPA */
+			memcpy(rl[idx].target_rpa.val, search_rpa->val,
+			       sizeof(bt_addr_t));
+			j = idx;
+		} else {
+			/* Could not resolve, and not in table */
+			j = FILTER_IDX_NONE;
+		}
+	}
+
+	/* Kick the callback in LLL (using the mayfly, tailchain it) */
+	if (twork->cb) {
+		mfy.fp = twork->cb;
+		mfy.param = (void *) ((unsigned int) j);
```

Comment:
There is no explanation of why param is j.
ku_resolve_callback_t should be documented along with any other things not obvious from the code .

Reply:
Comment added